src/toil_lib/pipelineWrapperBuilder.py

```diff
@@ -0,0 +1,160 @@
+    def run(self, args, pipeline_command):
+        """
+        Invokes the pipeline with the defined command. Command line arguments, and the command need
+        to be set with arg_builder, and command_builder respectively before this method can be
+        invoked.
+        """
+        # prepare workdir
+        mount = self._prepare_mount(args)
+        self._workdir = os.path.join(mount, 'Toil-' + self._name)
+        # prepare config
+        args_dict = vars(args)
+        args_dict['output_dir'] = mount
+        self._config = textwrap.dedent(self._config.format(**args_dict))
+        config_path = os.path.join(self._workdir, 'config')
+        command = self._make_command_prefix(os.path.join(self._workdir, 'jobStore'),
+                                            config_path,
+                                            self._workdir) + pipeline_command
+        if self._resume and args.resume:
+            command.append('--restart')
+        self._create_workdir(args)
+        with open(config_path, 'w') as f:
+            f.write(self._config)
+
+        try:
+            subprocess.check_call(command)
+        except subprocess.CalledProcessError as e:
+            print(e, file=sys.stderr)
+        finally:
+            log.info('Pipeline terminated, changing ownership of output files from root to user.')
+            stat = os.stat(self._mount)
+            subprocess.check_call(['chown', '-R', '{}:{}'.format(stat.st_uid, stat.st_gid),
+                                   self._mount])
+            if self._no_clean and args.no_clean:
+                log.info('Flag "--no-clean" was used, therefore %s was not deleted.', self._workdir)
+            else:
+                log.info('Cleaning up temporary directory: %s', self._workdir)
+                shutil.rmtree(self._workdir)
+
+    def get_args(self):
+        """
+        Use this context manager to add arguments to an argparse object with the add_argument
+        method. Arguments must be defined before the command is defined. Note that
+        no-clean and resume are added upon exit and should not be added in the context manager. For
+        more info about these default arguments see below.
+        """
+        parser = argparse.ArgumentParser(description=self._desc,
+                                         formatter_class=argparse.RawTextHelpFormatter)
+        # default args
+        if  self._no_clean:
+            parser.add_argument('--no-clean', action='store_true',
+                                help='If this flag is used, temporary work directory is not '
+                                     'cleaned.')
+        if self._resume:
+            parser.add_argument('--resume', action='store_true',
+                                help='If this flag is used, a previously uncleaned workflow in the'
+                                     ' same directory will be resumed')
+        return parser
+
+    def _prepare_mount(self, args):
+        assert args is not None
```

Comment:
`assert args is not None, 'No arguments provided, use the get_args method first.'`